### PR TITLE
front: add tests on useOutsideClick hook

### PR DESCRIPTION
**Pull request by theocrsb**

### CODE REVIEW

**Review by Math-R — COMMENTED** [review]
Works great but you have some linting errors to fix, ping me when you're ok

**Review by Maymanaf — COMMENTED** [review]
Im thinking of two more missing test cases: one for `ref = null `and one for cleanup on` unmount`

File: front/ui/ui-core/src/hooks/__tests__/useOutsideClick.spec.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import { renderHook } from '@testing-library/react';
+import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest';
+
+import useOutsideClick from '../useOutsideClick';
+
+describe('useOutsideClick', () => {
+  let div: HTMLDivElement;
+
+  beforeEach(() => {
+    div = document.createElement('div');
```
Comment (Maymanaf) [inline front/ui/ui-core/src/hooks/__tests__/useOutsideClick.spec.ts]: nit
```suggestion
    element = document.createElement('div');
```

File: front/ui/ui-core/src/hooks/__tests__/useOutsideClick.spec.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    document.body.appendChild(div);
+  });
+
+  afterEach(() => {
+    document.body.removeChild(div);
```
Comment (Maymanaf) [inline front/ui/ui-core/src/hooks/__tests__/useOutsideClick.spec.ts]: safer
```suggestion
   element.remove();
```

File: front/ui/ui-core/src/hooks/__tests__/useOutsideClick.spec.ts (hunk 1 above)
Comment (theocrsb) [reply front/ui/ui-core/src/hooks/__tests__/useOutsideClick.spec.ts]: done

File: front/ui/ui-core/src/hooks/__tests__/useOutsideClick.spec.ts (hunk 2 above)
Comment (theocrsb) [reply front/ui/ui-core/src/hooks/__tests__/useOutsideClick.spec.ts]: done

**Review by Maymanaf — APPROVED** [review]
Nice Work ✅

**Review by Math-R — APPROVED** [review]
Looks good to me ! thanks :)
